src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1151,55 +946,133 @@ public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Insta
+        Predicate hypervisorIdPredicate = criteriaBuilder.lower(hypervisorIdJoin
+            .get(HypervisorId_.hypervisorId)).in(lowerCaseHypervisorIds);
+
+        criteriaQuery.where(criteriaBuilder.and(ownerIdPredicate, hypervisorIdPredicate));
+
+        if (pageRequest != null) {
+            Path<String> sortBy = pageRequest.getSortBy() == null ?
+                hypervisorIdJoin.get(HypervisorId_.hypervisorId) :
+                root.get(pageRequest.getSortBy());
+            Order order = pageRequest.getOrder() == PageRequest.Order.DESCENDING ?
+                criteriaBuilder.desc(sortBy) :
+                criteriaBuilder.asc(sortBy);
+            criteriaQuery.orderBy(order);
+        }
+
+        TypedQuery<Consumer> query = this.getEntityManager().createQuery(criteriaQuery);
+
+        if (pageRequest != null && pageRequest.isPaging()) {
+            int offset = (pageRequest.getPage() - 1) * pageRequest.getPerPage();
+            int limit = pageRequest.getPerPage();
+            query.setFirstResult(offset).setMaxResults(limit);
+        }
+
+        return query.getResultList();
+    }
+
+    @Transactional
+    public Integer countHypervisorsBulk(String ownerId, Collection<String> hypervisorIds) {
+        if (hypervisorIds == null || hypervisorIds.isEmpty()) {
+            return 0;
+        }
+
+        List<String> lowerCaseHypervisorIds = toLowerCase(hypervisorIds);
 
         String jpql = """
-            SELECT c FROM Consumer c
+            SELECT COUNT(c) FROM Consumer c
             JOIN c.hypervisorId h
             WHERE c.ownerId = :ownerId
-                AND LOWER(h.hypervisorId) IN :lowerCaseHypervisorIds
-            ORDER BY h.hypervisorId ASC""";
+                AND LOWER(h.hypervisorId) IN (:lowerCaseHypervisorIds)
+            """;
 
-        List<Consumer> consumers = new ArrayList<>();
-        for (List<String> block : this.partition(lowerCaseHypervisorIds::iterator)) {
-            List<Consumer> result = getEntityManager().createQuery(jpql, Consumer.class)
-                .setParameter("ownerId", ownerId)
-                .setParameter("lowerCaseHypervisorIds", block)
-                .setLockMode(LockModeType.PESSIMISTIC_WRITE)
-                .getResultList();
+        // Query param limit handling was omitted. The maximum number of elements should be handled by
+        // pagination and so the limit should never be reached.
+        return getEntityManager().createQuery(jpql, Long.class)
+            .setParameter("ownerId", ownerId)
+            .setParameter("lowerCaseHypervisorIds", lowerCaseHypervisorIds)
+            .getSingleResult().intValue();
+    }
+
+    private static List<String> toLowerCase(Collection<String> hypervisorIds) {
+        List<String> lowerCaseHypervisorIds = hypervisorIds.stream()
+            .distinct()
+            .map(String::toLowerCase)
+            .sorted()
+            .toList();
+        return lowerCaseHypervisorIds;
```

Comment:
No need to for the `lowerCaseHypervisorIds` variable, we can just return the list:

```
    private static List<String> toLowerCase(Collection<String> hypervisorIds) {
        return hypervisorIds.stream()
            .distinct()
            .map(String::toLowerCase)
            .sorted()
            .toList();
    }
```